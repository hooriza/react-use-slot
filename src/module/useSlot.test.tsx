import renderer from "react-test-renderer";
import { useSlot } from ".";

test("FancyButton", () => {
  const FancyButton: React.FC = ({ children }) => {
    const Slot = useSlot(children);
    return (
      <button className="fancy-btn">
        <Slot />
      </button>
    );
  };

  expect(
    renderer.create(<FancyButton>Click me!</FancyButton>)
  ).toMatchSnapshot();
});

test("SubmitButton", () => {
  const SubmitButton: React.FC = ({ children }) => {
    const Slot = useSlot(children);
    return (
      <button type="submit">
        <Slot>Submit</Slot>
      </button>
    );
  };

  expect(renderer.create(<SubmitButton />)).toMatchSnapshot();
  expect(renderer.create(<SubmitButton>Save</SubmitButton>)).toMatchSnapshot();
});

test("SubmitButton", () => {
  const MyComponent: React.FC = ({ children }) => {
    const Slot = useSlot(children);

    return (
      <div className="my-component">
        <header>
          <Slot name="header" />
        </header>
        <main className="content">
          <Slot />
        </main>
        <footer>
          <Slot name="footer" />
        </footer>
      </div>
    );
  };

  expect(
    renderer.create(
      <MyComponent>
        <div slot="header">HEADER</div>
        <p slot="footer">FOOTER</p>
        CONTENT
      </MyComponent>
    )
  ).toMatchSnapshot();
});

test("List", () => {
  const List: React.FC = ({ children }) => {
    const Slot = useSlot(children);
    const dataList = ["foo", "bar", "baz"];

    return (
      <>
        <header>
          <Slot name="header" />
        </header>
        <ul>
          {dataList.map((data) => (
            <Slot name="item" />
          ))}
        </ul>
      </>
    );
  };

  expect(
    renderer.create(
      <List>
        <h1 slot="header">Title</h1>
        <li slot="item">How to show data??</li>
      </List>
    )
  ).toMatchSnapshot();
});

test("List with data", () => {
  const List: React.FC = ({ children }) => {
    const Slot = useSlot(children);
    const dataList = ["foo", "bar", "baz"];

    return (
      <>
        <header>
          <Slot name="header" />
        </header>
        <ul>
          {dataList.map((data) => (
            <Slot key={data} name="item" data={data} />
          ))}
        </ul>
      </>
    );
  };

  expect(
    renderer.create(
      <List>
        <h1 slot="header">Title</h1>
        <li slot="item">{(data: string) => data}</li>
      </List>
    )
  ).toMatchSnapshot();
});
