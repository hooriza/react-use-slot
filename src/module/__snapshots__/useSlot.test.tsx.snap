// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FancyButton 1`] = `
<button
  className="fancy-btn"
>
  Click me!
</button>
`;

exports[`List 1`] = `
Array [
  <header>
    <h1>
      Title
    </h1>
  </header>,
  <ul>
    <li>
      How to show data??
    </li>
    <li>
      How to show data??
    </li>
    <li>
      How to show data??
    </li>
  </ul>,
]
`;

exports[`List with data 1`] = `
Array [
  <header>
    <h1>
      Title
    </h1>
  </header>,
  <ul>
    <li>
      foo
    </li>
    <li>
      bar
    </li>
    <li>
      baz
    </li>
  </ul>,
]
`;

exports[`SubmitButton 1`] = `
<button
  type="submit"
>
  Submit
</button>
`;

exports[`SubmitButton 2`] = `
<button
  type="submit"
>
  Save
</button>
`;

exports[`SubmitButton 3`] = `
<div
  className="my-component"
>
  <header>
    <div>
      HEADER
    </div>
  </header>
  <main
    className="content"
  >
    CONTENT
  </main>
  <footer>
    <p>
      FOOTER
    </p>
  </footer>
</div>
`;
